examples/asset/texture_sampler.rs

```diff
@@ -0,0 +1,120 @@
+
+use bevy::app::App;
+use bevy::app::Startup;
+use bevy::asset::AssetServer;
+use bevy::asset::Assets;
+use bevy::math::Vec3;
+use bevy::prelude::*;
+use bevy::render::camera::ScalingMode;
+use bevy::render::mesh::Indices;
+use bevy::render::mesh::PrimitiveTopology;
+use bevy::render::texture::ImageAddressMode;
+use bevy::render::texture::ImageLoaderSettings;
+use bevy::render::texture::ImageSampler;
+use bevy::render::texture::ImageSamplerDescriptor;
+use bevy::sprite::MaterialMesh2dBundle;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+    asset_server: ResMut<AssetServer>,
+) {
+    commands.spawn(Camera2dBundle {
+        projection: OrthographicProjection {
+            scaling_mode: ScalingMode::AutoMin {
+                min_width: 2.,
+                min_height: 1.,
+            },
+            far: 1000.,
+            near: -1000.,
+            ..default()
+        },
+        ..default()
+    });
+
+    // Texture from ambientCG.com, licensed under the Creative Commons CC0 1.0 Universal License.
+    // https://ambientCG.com/a/Facade018A
+
+    // By default Bevy loads images to textures with sampler settings that clamp the image to the edges.
+    let image = asset_server.load("textures/facade018a.png");
+
+    // Here we override the sampler settings to repeat the image instead.
+    let image_repeat = asset_server.load_with_settings(
+        // We are using another file name, because Bevy ignores different loader settings for the same file.
+        // https://github.com/bevyengine/bevy/issues/11111
+        "textures/facade018a_copy.png",
+        |s: &mut ImageLoaderSettings| match &mut s.sampler {
+            ImageSampler::Default => {
+                s.sampler = ImageSampler::Descriptor(ImageSamplerDescriptor {
+                    address_mode_u: ImageAddressMode::Repeat,
+                    address_mode_v: ImageAddressMode::Repeat,
+                    ..default()
+                });
+            }
+            ImageSampler::Descriptor(sampler) => {
+                sampler.address_mode_u = ImageAddressMode::Repeat;
+                sampler.address_mode_v = ImageAddressMode::Repeat;
+            }
+        },
+    );
+
+    // Instead of using a standard quad with UV coordinates in the range `0..=1` (spanning the entire texture),
+    // we create a quad with UV coordinates in the range `-1..=2`, going beyond the texture by one entire unit in each direction.
+    // Texture behaviour then depends on the selected sampler mode.
+    // By default (`ImageAddressMode::ClampToEdge`) the out-of-bounds UV coordinates will map to the edge of the texture.
+    // When `ImageAddressMode::Repeat` is specified the out-of-bounds UV coordinates will repeat the texture.
```

Comment:
My opinion, which has very little weight in this context would be this:

Less text is better, because those who read the example, would have to spend less time and energy reading it.

> standard quad with UV coordinates

There's no such thing as "standard quad with UV coordinates". For example, if you have a mesh for a sphere, it usually contains hundreds quads, and none of them use coordinates 0..1. Quads with XY and UV coordinates 0..1 are used mostly in examples (and probably in Minecraft).

Also the comments below "usual UV range `0..=1`", "custom UV coordinates" are not exactly correct for the same reason.

> Texture behaviour then depends on the selected sampler mode

Texture behavior always depends on the sampler more, not "then". Perhaps you meant "Texture behaviour then depends on the selected **address** mode", which indeed *then* depends, because when UV is in range 0..=1, it does not depend on address mode.

> By default (`ImageAddressMode::ClampToEdge`) the out-of-bounds UV coordinates will map to the edge of the texture.
> When `ImageAddressMode::Repeat` is specified the out-of-bounds UV coordinates will repeat the texture

These would be the redundant comments because they
- repeat the mostly obvious enum variant name
- repeat enum documentation
- basically repeat what is said several lines above where the image is loaded (where this comments belongs)

Again, feel free to ignore this comment.

Reply:
You make a good point in regards to the **sampler** vs **address** mode, that should be reworded.

As per what a standard quad UV means, you're right that when you're talking about quad as a graphics primitive (part of a mesh), it could have any UVs for any reason, but when one talks about a quad as a mesh itself (a geometric quadrilateral) it does occur to me only as natural to use UV coordinates as usual.
I said usual UV range `0..=1`, because in that range lie UVs of most meshes people come into contact with (3D designers don't typically make their UVs sample outside of a texture from my experience).

You're right the comment does seem to give a bunch of redundant information that could be found in the docstrings for `ImageAddressMode`, though the purpose of the comment is mainly to explain why it takes user-handled in UV coordinates + a change in address mode to achieve the desired repeating effect, because just changing the adress mode would not actually produce any noticeable effects.
I'll consider shortening this part to a refference to the appropriate docs.

Anyway if someone gives me another opinion on this, I'll rewrite it.